AmazonKinesisVideoDemoApp/src/main/java/com/amazonaws/kinesisvideo/demoapp/fragment/StreamingFragment.java

```diff
@@ -135,18 +164,94 @@ private void pauseStreaming() {
         }
     }
 
+    private void updatePreviewSize(int width, int height) {
+        Log.d(TAG, "Choosing preview size...");
+        if (mShouldMaintainAspectRatio) {
+            cameraPreviewSize = chooseBestPreviewSize(width, height);
+        } else if (mRotation % 180 == 0) {
+            cameraPreviewSize = new Size(height, width);
+        } else {
+            cameraPreviewSize = new Size(width, height);
+        }
+        Log.d(TAG, "Preview size is: " + cameraPreviewSize.toString());
+        updateTransform(width, height);
+    }
+
+    private Size chooseBestPreviewSize(int width, int height) {
+        List<Size> allSizes = CameraHardwareCapabilitiesHelper.getSupportedResolutionsForYUV420_888(getContext(), mConfiguration.getCameraId());
+        List<Size> possibleSizes = new ArrayList<>();
+        for (Size possibility : allSizes) {
+            int longDimension = Math.max(width, height);
+            int shortDimension = Math.min(width, height);
+            if (possibility.getWidth() > longDimension && possibility.getHeight() > shortDimension) {
+                possibleSizes.add(possibility);
+            }
+        }
+        if (possibleSizes.size() >= 1) {
+            return Collections.min(possibleSizes, new Comparator<Size>() {
+                @Override
+                public int compare(Size s1, Size s2) {
+                    return Long.signum(s1.getWidth() * s1.getHeight() - s2.getWidth() * s2.getHeight());
+                }
+            });
+        }
+        return allSizes.get(0);
+    }
+
+    private void updateTransform(int width, int height) {
```

Comment:
can we add   description/ doc string on how this is used to fix rotation ?

Reply:
Done